### PR TITLE
feat[next]: enable gtir.embedded (and add support for Lists in output)

### PR DESCRIPTION
**Pull request by havogt**
Lists returned from an `as_fieldop`ed stencil will be turned into a local dimension of the the resulting field.
In case of a `make_const_list`, a magic local dimension `_CONST_DIM` is used. This is a hack, but localized to `itir.embedded`. A clean implementation will probably involve to tag the `make_const_list` with the neighborhood it is meant to be used with.

### CODE REVIEW

File: src/gt4py/next/ffront/foast_to_gtir.py [hunk 1]
```diff
@@ -116,7 +116,31 @@ def visit_FieldOperator(
     def visit_ScanOperator(
         self, node: foast.ScanOperator, **kwargs: Any
     ) -> itir.FunctionDefinition:
-        raise NotImplementedError("TODO")
+        # note: we don't need the axis here as this is handled by the program
```
Comment (havogt) [inline src/gt4py/next/ffront/foast_to_gtir.py:119]: copied from itir

File: src/gt4py/next/iterator/embedded.py [hunk 2]
```diff
@@ -920,9 +936,9 @@ def deref(self) -> Any:
         return _make_tuple(self.field, position, column_axis=self.column_axis)
 
 
-def _get_sparse_dimensions(axes: Sequence[common.Dimension]) -> list[Tag]:
+def _get_sparse_dimensions(axes: Sequence[common.Dimension]) -> list[common.Dimension]:
```
Comment (havogt) [inline src/gt4py/next/iterator/embedded.py:939]: just a cleanup

File: src/gt4py/next/iterator/type_system/type_specifications.py [hunk 3]
```diff
@@ -31,6 +31,7 @@ class OffsetLiteralType(ts.TypeSpec):
 @dataclasses.dataclass(frozen=True)
 class ListType(ts.DataType):
     element_type: ts.DataType
+    offset_type: Optional[common.Dimension] = None
```
Comment (havogt) [inline src/gt4py/next/iterator/type_system/type_specifications.py:34]: add a comment that it's not yet used in type inference

**Comment by havogt** [comment]
ready for a first round of review

File: src/gt4py/next/iterator/transforms/pass_manager.py [hunk 4]
```diff
@@ -223,10 +223,11 @@ def apply_fieldview_transforms(
 ) -> itir.Program:
     ir = inline_fundefs.InlineFundefs().visit(ir)
     ir = inline_fundefs.prune_unreferenced_fundefs(ir)
-    ir = InlineLambdas.apply(ir, opcount_preserving=True)
+    ir = InlineLambdas.apply(ir, opcount_preserving=True, force_inline_lambda_args=True)
     ir = infer_domain.infer_program(
         ir,
         offset_provider=offset_provider,
     )
+    # TODO move before infer_domain
```
Comment (tehrengruber) [inline src/gt4py/next/iterator/transforms/pass_manager.py:231]: Note: Blocked by https://github.com/GridTools/gt4py/pull/1706

File: tests/next_tests/unit_tests/iterator_tests/test_embedded_field_with_list.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
+# GT4Py - GridTools Framework
```
Comment (tehrengruber) [inline tests/next_tests/unit_tests/iterator_tests/test_embedded_field_with_list.py:1]: Maybe a skipped test for the case we don't support yet makes sense: `if(..., const_list_val, list_val)`

**Review by tehrengruber — APPROVED** [review]
As discussed I skipped over the details of the embedded implementation.

File: tests/next_tests/unit_tests/iterator_tests/test_embedded_field_with_list.py (hunk 5 above)
Comment (havogt) [reply tests/next_tests/unit_tests/iterator_tests/test_embedded_field_with_list.py:1]: done

**Pull request merged**